cli/src/commands/help.rs

```diff
@@ -42,8 +55,70 @@ pub(crate) fn cmd_help(
         .app()
         .clone()
         .subcommand_required(true)
+        .after_help(format_categories(command.app()))
         .try_get_matches_from(args_to_show_help)
         .expect_err("Clap library should return a DisplayHelp error in this context");
 
     Err(command_error::cli_error(help_err))
 }
+
+#[derive(Clone)]
+struct Category {
+    description: &'static str,
+    content: &'static str,
+}
+
+// TODO: Add all documentation to categories
+//
+// Maybe adding some code to build.rs to find all the docs files and build the
+// `CATEGORIES` at compile time.
+//
+// It would be cool to follow the docs hierarchy somehow.
+//
+// One of the problems would be `config.md`, as it has the same name as a
+// subcommand.
+//
+// TODO: Find a way to render markdown using ANSI escape codes.
+//
+// Maybe we can steal some ideas from https://github.com/martinvonz/jj/pull/3130
+const CATEGORIES: &[(&str, Category)] = &[
+    (
+        "revsets",
+        Category {
+            description: "A functional language for selecting a set of revision",
+            content: include_str!("../../../docs/revsets.md"),
+        },
+    ),
+    (
+        "tutorial",
+        Category {
+            description: "Show a tutorial to get started with jj",
+            content: include_str!("../../../docs/tutorial.md"),
+        },
+    ),
+];
+
+fn find_category(name: &str) -> Option<&Category> {
+    CATEGORIES
+        .iter()
+        .find(|(cat_name, _)| *cat_name == name)
+        .map(|tuple| &tuple.1)
+}
+
+fn format_categories(command: &clap::Command) -> String {
+    let subcommand_max_len = command
+        .get_subcommands()
+        .map(|cmd| cmd.get_name().len())
+        .max()
+        .unwrap();
```

Comment:
The `Categories`list doesn't have to be aligned with `Commands` list. I would use hard-coded width or max length of `cat_name`.